src/diffusers/schedulers/scheduling_dpmsolver_singlestep.py

```diff
@@ -597,6 +608,11 @@ def step(
         self.model_outputs[-1] = model_output
 
         order = self.order_list[step_index]
+
+        #  For img2img denoising might start with order>1 which is not possible
+        #  In this case make sure that the first two steps are both order=1
+        order = order if self.model_outputs[-order] is not None else order - 1
```

Comment:
`order` can be 1 to 3 if starts by random timestep.
So if you select timestep with order=3 as first this will fail, as will try order=2 which shouldn't be first.